lib/coinbase/stake.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+module Coinbase
+  # A representation of a Stake operation used for a end-user custody flow.
+  class Stake
+    class << self
+      # Builds a stake operation for a protocol on a given network.
+      # @param protocol [Symbol] The protocol name.
+      # @param network [Symbol] The network name.
+      # @param params  [Hash] The parameters needed to create a stake operation.
+      # @param opts [Hash] Optional parameters related to a stake operation.
+      # @return [String] An unsigned transaction used to stake.
+      def build_stake_operation(protocol, network, params, opts = { mode: :partial })
```

Comment:
I believe that these parameters should all be keyword parameters for our class methods.